### PR TITLE
DM-21355: Try to prevent NaN values turning up in FITS headers

### PR DESCRIPTION
**Pull request by timj**
The FITS standard doesn't allow NaN as a value. NaNs turning up in a PropertyList can confuse things so this change traps this and either drops the entire item or replaces with FITS undefined value.

### CODE REVIEW

File: src/fits.cc [hunk 1]
```diff
@@ -66,10 +66,28 @@ std::string makeLimitedFitsHeaderImpl(std::vector<std::string> const &paramNames
...
+            double value = metadata.get<double>(name);
+            if (!std::isnan(value)) {
+                // use G because FITS wants uppercase E for exponents
+                out += (boost::format("%#20.17G") % value).str();
+            } else {
+                LOGLS_WARN("afw.fits",
+                           boost::format("In %s, found NaN in metadata item '%s'") %
+                                         BOOST_CURRENT_FUNCTION % name);
+                // Convert it to FITS undefined
+                out += " ";
```
Comment (erykoff) [inline src/fits.cc]: Is this FITS undefined, or is it `"BLANK   "`? https://heasarc.gsfc.nasa.gov/docs/fcg/standard_dict.html
It could be that there's another translation layer somewhere else in the bowels of afw, or that I'm misreading something.

File: src/fits.cc [hunk 2]
```diff
@@ -66,10 +66,28 @@ std::string makeLimitedFitsHeaderImpl(std::vector<std::string> const &paramNames
...
+                out += (boost::format("%#20.15G") % value).str();
+            } else {
+                LOGLS_WARN("afw.fits",
+                           boost::format("In %s, found NaN in metadata item '%s'") %
+                                         BOOST_CURRENT_FUNCTION % name);
+                // Convert it to FITS undefined
+                out += " ";
+            }
         } else if (type == typeid(std::nullptr_t)) {
             out += " ";
```
Comment (erykoff) [inline src/fits.cc]: Hmmm, I guess this is what has always been used in `afw`, so I clearly don't have the full picture.

File: tests/test_frameSetUtils.py [hunk 3]
```diff
@@ -73,6 +73,18 @@ def getCrpix(self, metadata):
...
+        """
+        metadata = self.makeMetadata()
+        metadata["ISNAN"] = float("NaN")
+
+        # This will issue a warning from C++ but that warning can not be
+        # redirected to python logging machinery.
+        # This code causes a SEGV without DM-21355 implemented
+        frameSet1 = readFitsWcs(metadata, strip=False)
+        self.assertEqual(type(frameSet1), ast.FrameSet)
+
```
Comment (erykoff) [inline tests/test_frameSetUtils.py]: It makes sense that this test `SIGABRT`s on master, this was an untested edge case that could lead to an `SIGABRT` and it should now be tested (and fixed).

File: src/fits.cc (hunk 1 above)
Comment (timj) [reply src/fits.cc]: FITS undefined (there is a keyword and an `=` sign)

File: src/fits.cc (hunk 2 above)
Comment (timj) [reply src/fits.cc]: I added support for FITS undefined a few months back.
